gcn_prot/models/layers.py

```diff
@@ -41,3 +41,32 @@ def __repr__(self):
         return (
             f"{self.__class__.__name__} " f"({self.in_features} -> {self.out_features})"
         )
+
+
+class NormalizationLayer(nn.Module):
+    """Normalization layer for the adjacency matrix.
+
+    Adaped from https://github.com/CrivelliLab/Protein-Structure-DL/
+    """
+
+    def __init__(
+        self, in_features, out_features, dropout=0.1, bias=False, act=F.relu, D=100.0
+    ):
+        """Initialize layer."""
+        super(NormalizationLayer, self).__init__()
+        self.weight_feat = out_features
+        # Define trainable parameters
+        self.weight1 = nn.Linear(in_features, self.weight_feat, bias)
+        self.weight2 = nn.Linear(in_features, self.weight_feat, bias)
+        self.d = D
+
+    def forward(self, input):
+        """Normalize sparse adjacency matrix `adj` in terms of `v`."""
+        v, adj = input
+        v_shape = v.shape
+        if len(v_shape) > 2:
+            c1 = self.weight1(v)
+            c2 = self.weight2(v)
+            c = (F.sigmoid(c1 + c2) * self.d) + 0.00001  # As to not divide by zero
+            norm_adj = torch.exp(-((adj * adj) / (2 * c * c)))
+            return norm_adj
```

Comment:
The forward pass should retrieve both the unchanged `v` and `norm_adj`, because it need to pass both to the subsequent layers.